hypothesis-python/src/hypothesis/extra/pytestplugin.py

```diff
@@ -120,42 +124,36 @@ def pytest_runtest_makereport(item, call):
             'Hypothesis',
             '\n'.join(item.hypothesis_report_information)
         ))
+    if hasattr(item, 'hypothesis_statistics') and report.when == 'teardown':
+        # Running on pytest <= 3.5 where user_properties doesn't exist, fall
+        # back on the global gathered_statistics (which breaks under xdist)
+        if hasattr(report, 'user_properties'):  # pragma: no branch
+            report.user_properties += (
+                ('hypothesis-stats', item.hypothesis_statistics),
+            )
 
 
 def pytest_terminal_summary(terminalreporter):
     if not terminalreporter.config.getoption(PRINT_STATISTICS_OPTION):
         return
     terminalreporter.section('Hypothesis Statistics')
-    for name, statistics in gathered_statistics.items():
-        terminalreporter.write_line(name + ':')
-        terminalreporter.write_line('')
 
-        if not statistics.has_runs:
-            terminalreporter.write_line('  - Test was never run')
-            continue
+    if LooseVersion(pytest.__version__) < '3.5':  # pragma: no cover
+        if terminalreporter.config.getoption('-n'):
+            terminalreporter.write_line(
+                'Reporting Hypothesis statistics with pytest-xdist enabled '
+                'requires pytest >= 3.5'
+            )
+        for lines in gathered_statistics.values():
+            for li in lines:
+                terminalreporter.write_line(li)
+        return
 
-        terminalreporter.write_line((
-            '  - %d passing examples, %d failing examples,'
-            ' %d invalid examples') % (
-            statistics.passing_examples, statistics.failing_examples,
-            statistics.invalid_examples,
-        ))
-        terminalreporter.write_line(
-            '  - Typical runtimes: %s' % (statistics.runtimes,)
-        )
-        terminalreporter.write_line(
-            '  - Fraction of time spent in data generation: %s' % (
-                statistics.draw_time_percentage,))
-        terminalreporter.write_line(
-            '  - Stopped because %s' % (statistics.exit_reason,)
-        )
-        if statistics.events:
-            terminalreporter.write_line('  - Events:')
-            for event in statistics.events:
-                terminalreporter.write_line(
-                    '    * %s' % (event,)
-                )
-        terminalreporter.write_line('')
+    for test_report in terminalreporter.stats['']:
```

Comment:
What is `.stats['']` in this context?

Reply:
Magic.  It's the only place the list of `TestReport` objects seems to be stored; I copied this from an example plugin and it works even though I can't find where the entry is added to the dict.

I've added a comment which ~~will~~ would help me debug it ~~when~~ if this breaks later.  Fortunately we have tests that will fail immediately if it does!